bin/yb-ctl

```diff
@@ -1402,11 +1547,20 @@ class ClusterControl:
         self.creating_cluster = True
         server_counts = self.options.replication_factor
         self.set_master_addresses()
-        self.cluster_config = {
-            "enable_ysql": self.options.enable_ysql,
-            "num_drives": self.options.num_drives,
-            "replication_factor": self.options.replication_factor
-        }
+        self.cluster_config = {}
+        self.options.set_cluster_config(self.cluster_config)
+
+        for attr_name in (
+            "enable_ysql",
+            "num_drives",
+            "replication_factor"
+        ):
+            attr_value = getattr(self.options, attr_name)
+            if attr_value is not None:
+                self.cluster_config[attr_name] = attr_value
+        for protocol_type in PROTOCOL_TYPES:
+            port = self.options.base_ports[protocol_type]['rpc']
+            self.cluster_config[protocol_type + '_port'] = port
```

Comment:
this could just mimic the structure of the in memory dict, so we don't have to do _port here either